src/connection.rs

```diff
@@ -224,7 +226,17 @@ impl StreamMuxer for Connection {
         mut self: Pin<&mut Self>,
         cx: &mut Context<'_>,
     ) -> Poll<Result<StreamMuxerEvent, Self::Error>> {
+        let instant = Instant::now();
+
         while let Poll::Ready(Some(msg)) = self.inbound_rx.poll_recv(cx) {
+            // clear out pending data and return error if we've reached the timeout.
+            if instant.elapsed() >= self.poll_timeout {
+                self.pending_substreams.remove(&msg.substream_id);
+                self.pending_substream_data.remove(&msg.substream_id);
+
+                return Poll::Ready(Err(Error::ConnectionTimeout(self.poll_timeout)));
+            }
```

Comment:
this will only be triggered if we receive a message on the connection, but if we never receive a message (or receive `Data` immediately but never a `OpenResponse`) then the data will still be saved. I think we need to store a per-stream start time (maybe in the `pending_substreams` map?) and then each time poll it called, iterate through them and if any of them have reached the timeout, then delete it.

Reply:
@clearloop actually, don't worry about this, #18 fixes this also